src/components/Menu/MenuItem.tsx

```diff
@@ -86,17 +86,13 @@ class MenuItem extends UIComponent<any, any> {
     const { children, content } = this.props
 
     return (
-      <ElementType
-        className={classes.root}
-        onClick={this.handleClick}
-        {...accessibility.attributes.root}
-        {...rest}
-      >
+      <ElementType className={classes.root} {...accessibility.attributes.root} {...rest}>
         {childrenExist(children) ? (
           children
         ) : (
           <a
```

Comment:
Retested more keyboard navigation and readers.

I focus on space/enter functionality which should works on the menu item by this spec:
https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-1/menubar-1.html#

1. Enter/Space functionality without the readers:
- link without the "href" attribute doesn't invoke onclick event on enter/space key
- link with  href='javascript:void(0)' invoke onclick event on enter key. Spece key doesn't invoke onclick. 

2.  Enter/Space functionality with the readers:

2.1 Google/JAWS 
> if user navigate in virtual cursor mode then space/enter key invoke onclick event
> if user navigate to menu item by tab key then works the same as without screen reader(see section above)

2.2 Edge/Narrator 
> space/enter key invoke onclick event in scan mode or in normal navigation 

2.3 OSX/Ghrome/VoiceOver 
> space/enter key invoke onclick event when user press Control+Option+Space key.  Just enter/Space works the same as without screen reader(see section above)

2.4 Fifefox/NVDA
 > if user navigate in browse mode then space/enter key invoke onclick event
> if user navigate to menu item by tab key then works the same as without screen reader(see section above)

3. Then another option is replace link with button. Then tested space/enter key works in each combination. Without reader or with readers.  

**Note**: 
in aria example supporting space was done javascript:
    // Create simulated mouse event to mimic the behavior of ATs
    // and let the event handler handleClick do the housekeeping.
try {
          clickEvent = new MouseEvent('click', {
            'view': window,
            'bubbles': true,
            'cancelable': true
          });
        }
 catch (err) {
          if (document.createEvent) {
            // DOM Level 3 for IE 9+
            clickEvent = document.createEvent('MouseEvents');
            clickEvent.initEvent('click', true, true);
          }
        }
        tgt.dispatchEvent(clickEvent);
}